modules/ppcp-button/src/Assets/SmartButton.php

```diff
@@ -177,10 +177,6 @@ public function __construct(
 	 */
 	public function render_wrapper(): bool {
 
-		if ( ! $this->can_save_vault_token() && $this->has_subscriptions() ) {
```

Comment:
Why it was removed?

Reply:
@AlexP11223 

> make the subscriptions plugin create the customer before handling the payment
How can we do that?

Reply:
dunno, depends on how it does that. Maybe WC know something about that, or maybe they had issues like that before and there is a simpler solution?

Reply:
It does it by adding `createaccount` to the request :
```
	/**
	 * During the checkout process, force registration when the cart contains a subscription.
	 *
	 * @since 1.1
	 * @param $woocommerce_params This parameter is not used.
	 */
	public static function force_registration_during_checkout( $woocommerce_params ) {
		if ( WC_Subscriptions_Cart::cart_contains_subscription() && ! is_user_logged_in() ) {
			$_POST['createaccount'] = 1;
		}
	}
```

The problem I see is that we need the generated `data-client-token` added to JS script on page load:

![01](https://user-images.githubusercontent.com/456223/148082757-5dd818e5-3eff-4c6c-a4d4-12189e4ca9c7.png)

let´s imagine the guest is visiting a subscription product page, even if it´s technically possible to create an user programatically at this point, what happens if the customer leaves the site, the merchant is going to have a bunch of customers created in the system for no reason...

Reply:
In fact at this point not sure how is it even possible to create the customer as we don´t know the email, for example using:
`function wc_create_new_customer( $email, $username = '', $password = '', $args = array() ) {`

Reply:
hm, if we need it on page load, then yeah, I guess there is no simple way.

But then we should at least

> somehow extract/isolate the ID retrieval to avoid duplicating the user meta checks every time we need the ID.

and check that we did not miss any case. 

I think here we need this too?

https://github.com/woocommerce/woocommerce-paypal-payments/blob/21aa8980d642c0c3745d72b572e835aa4e97757e/modules/ppcp-subscription/src/RenewalHandler.php#L152

Reply:
@AlexP11223 I´ve extracted customer ID retrieval and reverted the conditional in `modules/ppcp-button/src/Assets/SmartButton::render_wrapper`.

Regarding cases where we need customer id from user, is only needed when requesting saved payments here: https://github.com/woocommerce/woocommerce-paypal-payments/blob/e9cafc762883beb62aa010a377590334b21c05a9/modules/ppcp-api-client/src/Endpoint/PaymentTokenEndpoint.php#L96

and when generating the client token here: https://github.com/woocommerce/woocommerce-paypal-payments/blob/e9cafc762883beb62aa010a377590334b21c05a9/modules/ppcp-api-client/src/Endpoint/IdentityToken.php#L108